### PR TITLE
[move-ide] Added support for dot-call syntax

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR adds support for dot-call syntax. In particular, `use fun` declarations are supported:

image

Also, method calls are supported:

image

 It required a few additions to the compiler, in particular:
- added `UseFuns` to `Sequence` and `ModuleDefinition` in the typed AST so that it's available to the IDE visitor
- added optional method name to `ModuleCall` in the typed AST so that we have it available when we are computing symbols over typed AST
- added `TypeName` to `UseFun` in the naming AST so that we can keep track of both `SomeStruct` locations in the following code snippet:
```
    public use fun Self::foo as SomeStruct.f1;
    public use fun Move2024::M1::foo as SomeStruct.f2;
```
-

## Test Plan 

New tests have been added.

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 1]
```diff
@@ -911,7 +911,9 @@ impl UseDefMap {
     }
 
     fn extend(&mut self, use_defs: BTreeMap<u32, BTreeSet<UseDef>>) {
-        self.0.extend(use_defs);
+        for (k, v) in use_defs {
```
Comment (awelc) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:914]: This was a bug that only manifested itself with this PR.

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 2]
```diff
@@ -1655,96 +1664,85 @@ impl<'a> ParsingSymbolicator<'a> {
...
+            P::ModuleUse::Members(v) => {
+                for (name, alias_opt) in v {
+                    self.use_decl_member_symbols(mod_defs, mod_ident_str.clone(), name, alias_opt);
+                }
+            }
+        }
+    }
+
+    /// Get symbols for a module member in the use declaration (can be a struct or a function)
+    fn use_decl_member_symbols(
```
Comment (awelc) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:1679]: This refactoring wasn't strictly necessary, but once done, it arguably makes the code look cleaner.

File: external-crates/move/crates/move-compiler/src/typing/translate.rs [hunk 3]
```diff
@@ -2230,7 +2232,17 @@ fn method_call(
         _ => exp_dotted_to_owned_value(context, DottedUsage::Use, loc, edotted, edotted_ty),
     };
     args.insert(0, first_arg);
-    let call = module_call_impl(context, loc, m, f, targs, parameters, argloc, args);
+    let call = module_call_impl(
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/src/typing/translate.rs:2235]: Might be easier to just set this when the call comes out?

**Review by tnowacki — COMMENTED** [review]
Also... macro expansion is going to ruin everything. Possibly. not sure what to do but we can improve on that overtime I suppose 

Just in that there will be no AST nodes for the macro itself, though we could try to add them

**Comment by awelc** [comment]
> Also... macro expansion is going to ruin everything. Possibly. not sure what to do but we can improve on that overtime I suppose

I was waiting for macros implementation to stabilize a bit before even thinking about it, but I guess it's about time...

We can still land this one, though, right? Meaning that macros are only going to ruin everything when they are actually used, rather than messing existing support up?

**Review by tnowacki — APPROVED** [review]
This looks easier than I would have thought to rebase on, so let's ship it!

**Comment by tnowacki** [comment]
> > Also... macro expansion is going to ruin everything. Possibly. not sure what to do but we can improve on that overtime I suppose
> 
> I was waiting for macros implementation to stabilize a bit before even thinking about it, but I guess it's about time...
> 
> We can still land this one, though, right? Meaning that macros are only going to ruin everything when they are actually used, rather than messing existing support up?

Correct yeah, just macros cannot be implemented easily with the existing strategy. So it might mean adding some more nodes in Typing about "Hey! There used to be a macro call here!" 
We thought about doing that anyway since it can help with error messages down the line

**Pull request merged**